torchtune/data/_messages.py

```diff
@@ -48,28 +46,40 @@ class Message:
             - For multiple consecutive assistant messages (i.e., tool calls
               by assistant), only the last assistant message will have ``eot=True``
             - All ipython messages (tool call returns) should set ``eot=False``.
+
+    Note:
+        Message class expects any image content to be in
+        `PIL Image format <https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image>`_.
     """
 
     def __init__(
         self,
         role: Role,
-        content: Union[str, List[Dict[str, str]]],
+        content: Union[str, List[Dict[str, Any]]],
         masked: bool = False,
         ipython: bool = False,
         eot: bool = True,
     ):
         self.role = role
-        self.content = (
-            [{"type": "text", "content": content}]
-            if isinstance(content, str)
-            else content
-        )
+        self.content = self._convert_to_list_of_dict(content)
         self.masked = masked
         self.ipython = ipython
         self.eot = eot
 
         self._validate_message()
 
+    def _convert_to_list_of_dict(self, content) -> List[Dict[str, Any]]:
+        """User is currently allowed to pass in a string for text-only content.
+        This ensures that the content is formatted as a list of dictionaries."""
+        if isinstance(content, str):
+            return [{"type": "text", "content": content}]
+
+        assert isinstance(
+            content, list
+        ), f"content must be of type List[Dict[str, Any]], got {content}"
```

Comment:
I know we don't wanna do it here, but is there somewhere that we can/should validate that we have a PIL image when type="image"?

Reply:
validate_messages?

Reply:
Argh, I cannot do this without a nested import of PIL.

Reply:
OK fine to leave it as is then